extension/page-detect.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+window.pageDetect = (() => {
+	const isGistCheck = location.hostname === 'gist.github.com';
+
+	const isGist = () => isGistCheck;
+
+	const isDashboard = () => location.pathname === '/' || /^(\/orgs\/[^\/]+)?\/dashboard/.test(location.pathname);
+
+	const isRepo = () => !isGist() && /^\/[^/]+\/[^/]+/.test(location.pathname);
```

Comment:
We could simplify a lot of these regexes by targeting them at `location.pathname` without user/repo. That way we could drop the first part of each regex for those.